tests/unit/viahtml/checkmate/client_test.py

```diff
@@ -0,0 +1,78 @@
+import pytest
```

Comment:
This whole business is intended to be totally separate from `viahtml` so we can pluck it off as a separate library when it's a bit more battle tested. The bigger this gets the less we want to copy paste it everywhere.

Some things it would be nice to add to this:

 * Helpers for handling any redirection we might want to issue from Checkmate
 * Maybe some helpers for filtering which kinds or responses you care about?
 * Maybe some caching on this side? No need to request the same URL over and over if we already know